moveit_core/macros/include/moveit/macros/declare_ptr.h

```diff
@@ -0,0 +1,55 @@
+/*********************************************************************
```

Comment:
It didn't seem logical to put something for declaring `shared_ptr` types in a file called `class_forward.h`. It has nothing to do with forward declaring a class, it just creates two type aliases. I suppose a different name could logically house both macros (though I can't think of one right now), but I'm guessing that would require keeping a backwards compatibility header for the old name for a while.

I figured this was the cleanest solution.